exist-core/src/main/java/org/exist/xquery/functions/xmldb/FunXCollection.java

```diff
@@ -21,37 +21,48 @@
  */
 package org.exist.xquery.functions.xmldb;
 
-//import org.apache.logging.log4j.LogManager;
-
-import org.exist.dom.QName;
-import org.exist.xquery.Cardinality;
 import org.exist.xquery.FunctionSignature;
+import org.exist.xquery.XPathException;
 import org.exist.xquery.XQueryContext;
 import org.exist.xquery.functions.fn.ExtCollection;
-import org.exist.xquery.value.FunctionReturnSequenceType;
-import org.exist.xquery.value.FunctionParameterSequenceType;
-import org.exist.xquery.value.SequenceType;
+import org.exist.xquery.value.Sequence;
 import org.exist.xquery.value.Type;
 
+import java.net.URI;
+
+import static org.exist.xquery.FunctionDSL.*;
+import static org.exist.xquery.functions.xmldb.XMLDBModule.functionSignature;
+
 /**
- * @author wolf
- * @author ljo
+ * @author <a href="mailto:adam@evolvedbinary.com">Adam Retter</a>
  */
 public class FunXCollection extends ExtCollection {
-//    private static final Logger logger = LogManager.getLogger(FunXCollection.class);
-	public final static FunctionSignature signature =
-        new FunctionSignature(
-            new QName("xcollection", XMLDBModule.NAMESPACE_URI, XMLDBModule.PREFIX),
-            "Returns the document nodes in the collections $collection-uris " +
-            "non-recursively, i.e. does not include document nodes found in " +
-            "sub-collections. " +
-            "C.f. fn:collection(). " +
-            XMLDBModule.COLLECTION_URI,
-            new SequenceType[] {
-		new FunctionParameterSequenceType("collection-uris", Type.STRING, Cardinality.ONE_OR_MORE, "The collection URIs")},
-            new FunctionReturnSequenceType(Type.NODE, Cardinality.ZERO_OR_MORE, "the document nodes from the specified collections excluding sub-collections"));
-
-	public FunXCollection(XQueryContext context) {
-		super(context, signature, false);
+
+	private static final String FS_XCOLLECTION_NAME = "xcollection";
+	static final FunctionSignature FS_XCOLLECTION = functionSignature(
+			FS_XCOLLECTION_NAME,
+			"Returns the document nodes in the collections $collection-uris " +
+					"non-recursively, i.e. does not include document nodes found in " +
+					"sub-collections. " +
+					"C.f. fn:collection().",
+			returnsOptMany(Type.ITEM, "The items from the specified collections excluding sub-collections"),
+			params(
+				manyParam("args", Type.STRING,"The Collection URI")
```

Comment:
The very generic argument name `args` should either stay `collection-uris` or use `uris` in line with the XQ40 spec.

Reply:
eXist-db targets XQuery 3.1 not 4.0 (yet).
The official 3.1 spec compliant function `fn:collection` uses the term `arg` as this is a pluralised version of that function, using `args` here makes sense.

Reply:
`xmldb:xcollection` is not part of any XQuery spec.